torch/distributed/_shard/api.py

```diff
@@ -87,6 +90,11 @@ def shard_parameter(
     This method replaces ``module.param_name`` with a
     :class:`torch.distributed._sharded_tensor.ShardedTensor`
 
+    If the parameter is a MetaTensor, then an empty ShardedTensor is
+    materialized as per the provided ``sharding_spec`` and ``src_rank`` is ignored.
+    In this case, it is the user's responsibility to appropriately initialize
```

Comment:
Could the docs add an example of how to do this? Is it with the usual `reset_parameters`?